### PR TITLE
Added md5 message digest function to encrypt strings.

### PR DESCRIPTION
**Pull request by EricDDK**
This is a function of encrypting strings, generally used for message digests and checksum data.

### CODE REVIEW

File: cocos/base/ccUtils.h [hunk 1]
```diff
@@ -173,6 +173,12 @@ namespace utils
      */
     CC_DLL std::string getFileMD5Hash(const std::string &filename);
 
+	/**
+	*  Gets the md5 hash for the given string.
+	*  @param the string to calculate md5 hash.
+	*  @return The md5 hash for the string
+	*/
+	CC_DLL std::string getStringMD5Hash(const std::string &dataStr);
```
Comment (PatriceJiang) [inline cocos/base/ccUtils.h:181]: Please fix the code alignment

**Comment by PatriceJiang** [comment]
Could you please rebase the change on to the recent v3 branch since the commit log is not brief.

screen shot 2019-01-04 at 3 02 11 pm

File: cocos/ui/UIImageView.h [hunk 2]
```diff
@@ -118,6 +118,12 @@ class CC_GUI_DLL ImageView : public Widget , public cocos2d::BlendProtocol
      * @see `setCapInsets(const Rect&)`
      */
     const Rect& getCapInsets()const;
+
+	/**
+	* Get ImageView's texture file.
+	* @return texture file.
+	*/
+	const std::string getTextureFile()const;
```
Comment (Dimon4eg) [inline cocos/ui/UIImageView.h:126]: better to return const reference not just const value

**Review by Dimon4eg — CHANGES_REQUESTED** [review]

File: cocos/ui/UIImageView.cpp [hunk 3]
```diff
@@ -238,6 +238,11 @@ const Rect& ImageView::getCapInsets()const
     return _capInsets;
 }
 
+const std::string ImageView::getTextureFile()const
```
Comment (halx99) [inline cocos/ui/UIImageView.cpp:241]: No need const qualifier for non-reference result type.

**Pull request closed without merging**